packages/react-dom/src/__tests__/ReactUpdates-test.js

```diff
@@ -870,27 +870,12 @@ describe('ReactUpdates', () => {
       );
     }
     component = ReactTestUtils.renderIntoDocument(<A />);
-    expect(() => component.setState({}, {foo: 'bar'})).toThrowError(
```

Comment:
This actually removes valid assertions. :-) Yes, here they get deduplicated, but we still want to test those cases. Maybe let's use both caller name and the stringified "received" as key (combined)?

Reply:
Yeah you're right. For now, `${callerName}_${JSON.stringify(callback)}` is the key of `didWarnOnInvalidCallback`.

Reply:
Shall I close this pull request due to no response? @gaearon

Reply:
Please don't expect fast responses during the holidays :-)